lib/fizzy/execute.cpp

```diff
@@ -172,11 +172,27 @@ Instance instantiate(const Module& module)
     }
     // NOTE: fill it with zeroes
     bytes memory(memory_min * page_size, 0);
-    return {module, std::move(memory), memory_max};
+
+    // TODO: add imported globals first
+    std::vector<uint64_t> globals;
+    globals.reserve(module.globalsec.size());
```

Comment:
Can't you just use `std::vector<uint64_t> globals(module.globalsec.size())` ?

Reply:
there's no constructor that reserves but not resizes